editor.planx.uk/src/components/Feedback/MoreInfoFeedback.test.tsx

```diff
@@ -0,0 +1,155 @@
+    insertFeedbackMutation: jest.fn(),
+  };
+});
+
+const scrollIntoView = jest.fn();
+window.Element.prototype.scrollIntoView = scrollIntoView;
+
+describe("MoreInfoFeedbackComponent presentation and functionality", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  // Initial load
+  test('Initial loads renders "Yes" and "No" buttons initially', () => {
+    const { getByText } = setup(<MoreInfoFeedbackComponent />);
+    expect(getByText("Yes")).toBeInTheDocument();
+    expect(getByText("No")).toBeInTheDocument();
+  });
+
+  test("Does not scroll into view on initial render", () => {
+    setup(<MoreInfoFeedbackComponent />);
+    expect(scrollIntoView).not.toBeCalled();
+  });
+
+  // Sentiment selection
+  test("Clicking Yes input form scrolls into view", async () => {
+    const { getByText } = setup(<MoreInfoFeedbackComponent />);
+    fireEvent.click(getByText("Yes"));
+    expect(scrollIntoView).toBeCalled();
+    await waitFor(() => {
+      expect(
+        getByText("Please help us to improve this service by sharing feedback"),
+      ).toBeInTheDocument();
+    });
+  });
+
+  test("Clicking No input form scrolls into view", async () => {
+    const { getByText } = setup(<MoreInfoFeedbackComponent />);
+    fireEvent.click(getByText("No"));
+    expect(scrollIntoView).toBeCalled();
+    await waitFor(() => {
+      expect(
+        getByText("Please help us to improve this service by sharing feedback"),
+      ).toBeInTheDocument();
+    });
+  });
+
+  // Form submission
+  test("Submitting feedback changes view to thank you message", async () => {
+    const { getByText, getByTestId } = setup(<MoreInfoFeedbackComponent />);
+
+    fireEvent.click(getByText("Yes"));
+    await waitFor(() => {
+      expect(getByTestId("userCommentTextarea")).toBeInTheDocument();
+    });
+
+    await userEvent.type(
+      getByTestId("userCommentTextarea"),
+      "Great help, thanks!",
+    );
+
+    fireEvent.click(getByText("Send feedback"));
+    await waitFor(() => {
+      expect(getInternalFeedbackMetadata).toBeCalled();
+      expect(insertFeedbackMutation).toBeCalled();
+    });
+
+    await waitFor(() => {
+      expect(getByText("Thank you for your feedback.")).toBeInTheDocument();
+    });
+  });
+
+  /*
+    We use the `required` property to validate that a user can't submit an empty
+    comment. 
+    
+    It doesn't seem to be possible to test that the Browser stops form
+    submit in the Jest environment.
+    
+    Checking for `required` property currently but we could add explicit
+    validation.
+  */
+  test("Feedback form requires a comment before submitting", async () => {
+    const { getByTestId, getByText } = setup(<MoreInfoFeedbackComponent />);
+
+    fireEvent.click(getByText("Yes"));
+    await waitFor(() => {
+      expect(getByTestId("userCommentTextarea")).toBeInTheDocument();
+    });
+
+    expect(getByTestId("userCommentTextarea")).toHaveAttribute("required");
+  });
```

Comment:
This test highlights something interesting - while we can't test that the browser stops form submit in Jest, we'd normally test that an input error message becomes visible when trying to proceed without filling out required fields  - but it doesn't seem like we've accounted for that with any of these components and are relying on default browser behavior? 

I'd expect us to the use the same GOV.UK pattern of red left border and message that we use for all other public-facing inputs - is there a reason we aren't? I think it's very likely this will be picked up in a11y testing as inconsistent and insufficient.

![Screenshot from 2024-02-23 15-43-21](https://github.com/theopensystemslab/planx-new/assets/5132349/00652bc3-fb88-4b4f-920e-2f8755eb4903)

![Screenshot from 2024-02-23 15-36-07](https://github.com/theopensystemslab/planx-new/assets/5132349/8109f08d-7831-469f-a4af-0f499633f907)

Reply:
At the time this feature was being built I think there were conversations about using the native `required` although to be fair that might have been scoped to the editor rather than the public view.

I did have a version with the gov uk style errors but they could never actually be seen by users as the `required` attribute stopped form submission which would be run to show these validation messages so that state could never be reached. 

As this was the case at the time I thought I'd reduce complexity but not showing any code which I thought a user could never see. 

It'll be really interesting to see what the a11y testing comes back with because on one hand not using `required` for a required field seems somewhat off? Although as you say it's inconsistent 🤔